custom/icds_reports/tests/agg_tests/test_aggregation_script.py

```diff
@@ -432,16 +432,10 @@ def test_aggregate_query(self):
         with get_cursor(AggregateInactiveAWW) as cursor:
             cursor.execute(missing_location_query)
             cursor.execute(aggregation_query, agg_params)
+
```

Comment:
I'm fine with this change, but I'm cautious about combining tests.

There is a belief that tests should test only one thing (even if you use multiple `assert`s to test it). If you are testing multiple things in the same test, and an early assert fails, then the test never reaches later asserts and you can't see whether one thing is broken or many things. Of course debugging is easier if you know exactly what is broken and what isn't -- how much time have I spent on Interrupt thinking the problem was one thing when actually :man_facepalming: it was a completely different thing?

([Context managers are very cool](https://github.com/dimagi/commcare-hq/blob/eee7291d96d77cd941dc734dd8da99482b3de0ad/corehq/util/test_utils.py#L455) for keeping setup and teardown code together, and for reusing it anywhere you want. ... :trollface: I think the only reason JUnit (which Python's unittest library is based on) has setUp() and tearDown() methods is because, prior to Java 7, Java didn't have [anything like context managers](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html), and it can't use them as decorators.)

Reply:
Thanks for the notes, Norman! Very valid about combining tests. In this case, combining would result in 6 SQL queries instead of 12 sql queries, and the only difference in the combined tests is that one looks at the number of rows and the other one looks at a specific row, so I thought it made since to combine in this case.